msdm/core/distributions/distributions.py

```diff
@@ -91,5 +91,17 @@ def isclose(self, other: "FiniteDistribution[Event]") -> bool:
                 return False
         return True
 
+    def marginalize(self, projection: Callable[[Event], Event]):
+        newdist = defaultdict(lambda : 0)
+        for e, p in self.items():
+            newdist[projection(e)] += p
+        return DictDistribution(newdist)
+
+    def expectation(self, real_function: Callable[[Event], float]):
```

Comment:
could have this default to the identity function if you want to get mean for free?